lib/spack/spack/test/cmd/is_git_repo.py

```diff
@@ -42,7 +45,13 @@ def git_tmp_worktree(tmpdir):
     """Create new worktree in a temporary folder and monkeypatch
     spack.paths.prefix to point to it.
     """
-    worktree_root = str(tmpdir.join("tmp_worktree"))
+    if sys.platform == 'win32':
+        long_pth = str(tmpdir).split(os.path.sep)
+        tmp_worktree = os.path.sep.join(long_pth[:-1])
```

Comment:
It looks like this is trying to move to the parent directory of `tmpdir` - does something break if you just work inside `tmpdir` itself?

Reply:
We run into Windows path length issues unfortunately.

Reply:
I was thinking this would potentially be fragile: if the path is too long and we strip off one directory, I feel like things will be close to breaking again.

I don't see any good solutions for that though (e.g. `https://github.com/pytest-dev/pytest/issues/5802`). It looks like `spack.cmd.unit_test` falls back to a vendored (and old) version of `pytest` if we can't import it - do you know if we are using the vendored version or not on Windows? It might be that a newer version would avoid this issue.